src/DocTests.jl

```diff
@@ -124,7 +124,11 @@ function doctest(ctx::DocTestContext, block_immutable::Markdown2.CodeBlock)
     if startswith(lang, "jldoctest")
         # Define new module or reuse an old one from this page if we have a named doctest.
         name = match(r"jldoctest[ ]?(.*)$", split(lang, ';', limit = 2)[1])[1]
-        sym = isempty(name) ? gensym("doctest-") : Symbol("doctest-", name)
+        if isempty(name)
+            sym = Symbol("__doctest__", filter(!=('#'), string(gensym())))
+        else
+            sym = Symbol("__doctest__", name)
```

Comment:
There is a theoretical possibility of the generated symbol conflicting with a named doctest (e.g. if someone uses `260` or something like that as their doctest name). This should get around that:

```suggestion
            sym = Symbol("__doctest__named__", name)
```